src/Microsoft.AspNetCore.SignalR.Core/Internal/DefaultHubDispatcher.cs

```diff
@@ -88,7 +88,8 @@ public partial class DefaultHubDispatcher<THub> : HubDispatcher<THub> where THub
 
                 case StreamInvocationMessage streamInvocationMessage:
                     Log.ReceivedStreamHubInvocation(_logger, streamInvocationMessage);
-                    await ProcessInvocation(connection, streamInvocationMessage, isStreamedInvocation: true);
+                    // Fire-and-forget stream invocations, otherwise they would block other hub invocations from being able to run
+                    _ = ProcessInvocation(connection, streamInvocationMessage, isStreamedInvocation: true);
```

Comment:
I mostly agree with @JamesNK here. Though to be fair, this isn't a full defense against bad code, there's still nothing stopping a poorly-written Hub method from `Task.Run`ing stuff. Still, if we `await` invocation of the method and then only fire-and-forget the logic to drain the stream, it would fit user expectations more I think.